internal/driver/config/provider.go

```diff
@@ -155,23 +156,39 @@ func (k *Config) Set(key string, v any) error {
 func (k *Config) ReadAPIListenOn() string {
 	return fmt.Sprintf(
 		"%s:%d",
-		k.p.StringF(KeyReadAPIHost, ""),
-		k.p.IntF(KeyReadAPIPort, 4466),
+		k.p.String(KeyReadAPIHost),
+		k.p.Int(KeyReadAPIPort),
 	)
 }
 
-func (k *Config) MaxReadDepth() int {
-	return k.p.Int(KeyLimitMaxReadDepth)
+func (k *Config) WriteAPIListenOn() string {
+	return fmt.Sprintf(
+		"%s:%d",
+		k.p.String(KeyWriteAPIHost),
+		k.p.Int(KeyWriteAPIPort),
+	)
 }
 
-func (k *Config) WriteAPIListenOn() string {
+func (k *Config) MetricsListenOn() string {
+	return fmt.Sprintf(
+		"%s:%d",
+		k.p.String(KeyMetricsHost),
+		k.p.Int(KeyMetricsPort),
+	)
+}
```

Comment:
Maybe instead we could do

```suggestion
func (k *Config) listenOn(hostKey, portKey string) string {
	return fmt.Sprintf(
		"%s:%d",
		k.p.String(hostKey),
		k.p.Int(portKey),
	)
}

func (k *Config) MetricsListenOn() string {
	return k.listenOn(KeyMetricsHost, KeyMetricsPort)
}
```

Reply:
Yes, I'll refactor. That part was nagging me as well :)